src/shared/install.c

```diff
@@ -2426,32 +2457,75 @@ int unit_file_query_preset(UnitFileScope scope, const char *root_dir, const char
                                 continue;
 
                         parameter = first_word(l, "enable");
-                        if (parameter) {
-                                if (fnmatch(parameter, name, FNM_NOESCAPE) == 0) {
-                                        log_debug("Preset file says enable %s.", name);
-                                        return 1;
-                                }
-
-                                continue;
-                        }
+                        if (parameter)
+                                rule = (PresetRule) {
+                                        .pattern = strdup(parameter),
```

Comment:
yeah, @msekletar i'd always try to keep the operatoin and the following error checking close. It's a pattern that is just easier to follow.